### PR TITLE
✨ Added more features to the restful API

### PR DESCRIPTION
**Pull request by KillerBOSS2019**
Added `skip-ad` No value needed
Added `start-playlist` with playlist name value required

### CODE REVIEW

**Comment by KillerBOSS2019** [comment]
```py
import requests
requests.post("http://localhost:9863/query", json={"command": "play-url", "value": "https://music.youtube.com/watch?v=kzxr3NKqV_c&list=OLAK5uy_mojCSnGfQeUmoYsBOxsPO2Uzz9ioBqZtU"})
```

**Comment by KillerBOSS2019** [comment]
image
Added Auto Skip ad when Skip-ad button is available

**Comment by Alipoodle** [comment]
😅 Have you seen my Reviews?

**Comment by KillerBOSS2019** [comment]
Hm I don't see the review

**Review by Alipoodle — CHANGES_REQUESTED** [review]
😋 Finally found what I needed to do before submitting the full review.

Yea, just a few changes to try and make it a bit more inline with the rest of YTM Desktop.

File: src/providers/infoPlayerProvider.js [hunk 1]
```diff
@@ -450,6 +481,64 @@ function setVolume(webContents, time) {
...
+        )
+        .then()
+        .catch((_) => console.log('error Playlist ' + _))
+}
+
+function playURL(url) {
+    if (url && url.includes('watch?v=')) {
+        let video = url.split('watch?v=')
+        let baseurl = 'ytmd://play/'
+        if (video[0] == 'https://music.youtube.com/' && false) {
```
[the remark this answers is not in the document]
Comment (KillerBOSS2019) [reply src/providers/infoPlayerProvider.js:520]: Oh woops I forgot to remove that. It was there for testing

File: src/pages/settings/settings.html [hunk 2]
```diff
@@ -539,6 +585,24 @@
                                             </div>
                                         </td>
                                     </tr>
+                                    <tr>
+                                        <td>
+                                            <span
+                                                i18n="i18n_LABEL_SETTINGS_TAB_PLAYBACK_AutoClick_SkipAd"
```
[the remark this answers is not in the document]
Comment (KillerBOSS2019) [reply src/pages/settings/settings.html]: https://github.com/ytmdesktop/ytmdesktop-locales/pull/13

File: src/pages/settings/settings.html [hunk 3]
```diff
@@ -599,15 +599,24 @@
                                     <tr>
                                         <td>
                                             <span
-                                                i18n="i18n_LABEL_SETTINGS_TAB_PLAYBACK_AutoClick_SkipAd"
+                                                i18n="i18n_LABEL_SETTINGS_TAB_PLAYBACK_AUTOCLICK_SKIPAD"
                                             ></span>
+                                                i18n="i18n_LABEL_SETTINGS_TAB_PLAYBACK_SURROUND_SOUND"
```
Comment (Alipoodle) [inline src/pages/settings/settings.html:604]: 🤔 I think some issues have occurred here with the merging of a few PRs last night for the Surrounds Sound.
Looks like they've both overlapped each other as values in the settings.

**Comment by Alipoodle** [comment]
Great to see the PR in Locale & The rest of the changes done.
I'll give this a test for next week 😄, I've got 10 days free which I plan to go through some PRs on here!

File: src/pages/settings/settings.html (hunk 3 above)
Comment (KillerBOSS2019) [reply src/pages/settings/settings.html:604]: Noticed I'll fix it when I have time

**Comment by Alipoodle** [comment]
Strangely enough I was having issue in regards to getting it to re-load the page for a URL Link while in the dev launch option. 🤷‍♂️
Couldn't figure out why, but this may need to be looked into.
I've had to fix the handleOpenURL as this was broken in both Dev & Normal run.

Not 100% happy with the loading of Playlists as I feel that running the URL of: `https://music.youtube.com/library/playlists` and once it's completed (with a .then()) we pick the playlist.
Otherwise some people may wish to provide a Playlist ID?

Skip-ad I've been unable to test otherwise, however I feel that this is at least a good enough start with my fixes to be added.

**Comment by KillerBOSS2019** [comment]
any updates on this?

**Comment by KillerBOSS2019** [comment]
So I fixed the problem where play-url don't work. and I changed so you will need a playlist id in order to start your playlist

**Comment by KillerBOSS2019** [comment]
Can this be merged? I know a lot of people who use Stream Deck or TouchPortal are very excited this PR

**Comment by KillerBOSS2019** [comment]
`query/playlist` endpoint always gives empty playlist. because YTM has updated their path id. this PR also updated the path so now the endpoint should work again.

**Comment by Alipoodle** [comment]
Closing this PR in the move to YTMD v2

**Pull request closed without merging**